### ISSUE TITLE
wagtail.contrib.modeladmin resorts to defaults

### ISSUE THREAD

**Issue by rcrowther** [wagtail/wagtail#5963]
Found a bug? Please fill out the sections below. 👍

### Issue Summary

From documentation, it seems that displaying Django models in Wagtail admin has been a concern in the past, to the point where Wagtail includes an app.

http://docs.wagtail.io/en/latest/reference/contrib/modeladmin/index.html

Unfortunately, documentation falls short on my scenario. In my scenario enabling the app with modern model designs produced admin forms with only default field rendering. Configuration had no effect. 

And then I found a solution, using overriding on the app referenced above, wagtail.contrib.modeladmin.

### Steps to Reproduce

1. Form a standard Wagtail model with StreamField and inage fields 
2. Add panels
```
    content_panels = Page.content_panels + [
        FieldPanel('rdate'),
        ImageChooserPanel('img'),
        StreamFieldPanel('text'),
    ]
```
Now you decide you don't want this model under Wagtail.

3. Convert the Wagtail model to a plain Django model
`
    class SomeClassname(models.Model):
`
Maybe add a title field.

4. Fit Django Admin
In a file,
```
    class ArticleAdmin(admin.ModelAdmin):
        pass
    
    admin.site.register(Review, ReviewAdmin)
```
Admin appears, but the Wagtail speciality fields look raw. Image upload reverted to a Select button, and StreamField has an interface that has clearly lost it's way.

5. Install wagtail.contrib.modeladmin
Settings, then a wagtail_hook.py file,
```
    class ArticleAdmin(ModelAdmin):
        pass

    modeladmin_register(ReviewAdmin)
```

6. Install wagtail.contrib.modeladmin
Settings, then a wagtail_hook.py file,
```
    class ArticleAdmin(ModelAdmin):
        pass

    modeladmin_register(ReviewAdmin)
```
Forms for the new model now appear in Wagtail admin. But with some weird combo of gadgets that look like Django widgets. Yet looking at the HTML underneath, a different template.

7. Try configure the admin in wagtail_hook.py
Something like,
```

    class ArticleAdmin(ModelAdmin):
        fields = ('make', 'mdl', 'img')

    modeladmin_register(ReviewAdmin)
```
Nothing. No effect from any configuration. Same form result.

8. Move the panels definition to wagtail_hook.py and rename as attribute ''panels'
Mashing code together,
```
    class ArticleAdmin(ModelAdmin):
        panels = Page.content_panels + [
            FieldPanel('rdate'),
            ImageChooserPanel('img'),
            StreamFieldPanel('text'),
        ]

    modeladmin_register(ReviewAdmin)
```
Now Wagtail admin shows the model forms with Wagtail widgets.

###  Expected behaviour
From a breif note in documentation, and the claims of the original module, I gained a hope that a model writen in Django could be supported through Wagtail admin. I even expected the resulting admin to support the use of Wagtail specific fields, using Wgtail widgets.

### Actual behaviour
The bridge module wagtail.contrib.modeladmin suceeded in displaying an interface in Wagtail admin, but failed to cope with Wagtail specific fields. It displayed a muddle that seemed to derive from defaults, and was on Streamfield unusable.

### Testing depth
I am in no position to test this widely, or hack about.  I can only confirm it happened, and that Django admin, which I ran concurrently, did respond to configuration.

### Further Notes
This is related to,
https://github.com/wagtail/wagtail/issues/2717
but now Wagtail is, I think, showing a further (and unnecessary?) loss of functionality, possibly due to development in panel definitions?

There are some odd features of this scenario. If the bridge module was placed in Wagtail permanently, it must have had some usage and demand to be placed there. Yet I've looked back through previous bug reports and found no mention.

Documentation suggests this should be possible. The fix suggests not only should it be possible, it may once have been routine.

If this is a functionality that should work, and is not, tracing may be a little difficult without a dedicated environments. The code of the ''option' method and admin in general at this point is dense, and Python likes to float untyped sets of arguments about. On the other hand ''panels' as opposed to ''content_panels' should give an experienced developer a massive head start.

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (yes , with regrets for my inability to chase the issue)

### Technical details

* Python version: 3.82 
* Django version:  3.0.5 
* Wagtail version: 2.8.1 
* Browser version: Firefox 74

**Comment by gasman** [comment]
Hi @rcrowther,
It's a little hard to follow your code examples, as you've only provided partial class definitions and the class names are changing from one step to the next. However, it sounds like you've misunderstood the purpose of wagtail.contrib.modeladmin - it is not intended to be a bridge between the Django admin and Wagtail admin, but simply a way of making arbitrary models available within the Wagtail admin. Wagtail does not provide any capabilities for customising the behaviour of the Django admin interface.

I'm afraid I can't make sense of your expected vs actual behaviour - at the end of step 8 you say you have it working within the Wagtail admin, but then go on to say that it doesn't work. If it's the latter, please open a new issue with a complete runnable code example that demonstrates the problem.
